### ISSUE TITLE
Set grid-max based on values in displayed interval

### ISSUE THREAD

**Issue by deridex** [grafana/grafana#2548]
It seems that a chart grid-max is set to the lifetime max of a measurement value. We are storing the duration of an operation and had a few outliers. Now, the y-axis of the duration chart is permanently (as far as I can tell) scaled to accommodate those outliers, even though the time interval currently being displayed contains, by comparison, tiny values. We have to manually set the grid-max to get any benefit from the chart.

With grid-max auto and all data lines squished together:

![auto](https://cloud.githubusercontent.com/assets/9098/9338733/36b20270-45b4-11e5-912d-c3aff544f5a9.png)

With explicit grid-max so we get separation between the lines:

![explicit](https://cloud.githubusercontent.com/assets/9098/9338748/43973fd2-45b4-11e5-9625-0dd7ac58d9fc.png)

Out data source is influx.

**Comment by torkelo** [comment]
I cant see any outliers in the above graph, can you show me the query and the data return by influxdb?

**Comment by torkelo** [comment]
you find the raw data returned by influxdb in the chrome dev tools (f12) network tab

**Comment by deridex** [comment]
Fixed it. Total user error. Rather than using the query builder widget, which groups by time interval and constrains by time, I had manually entered a query, which does neither. As a result, the manual query returned _everything_ and then rendered only the datapoints that fell into the displayed time interval.

Now that I am using the query builder, the results are constrained to just the displayed time interval and the grid-max auto scale does sets the appropriate max y value.
